core/api/test/unit/domain/bitcoin/onchain/checked-to-on-chain-address.spec.ts

```diff
@@ -2,19 +2,36 @@ import { BtcNetwork } from "@/domain/bitcoin"
 import { checkedToOnChainAddress } from "@/domain/bitcoin/onchain"
 import { InvalidOnChainAddress } from "@/domain/errors"
 
-const addresses = [
+const validAddresses = [
   { network: "mainnet", address: "17sELMebjQNf1k1CRucF67QAtegNsTjXUn" },
   { network: "mainnet", address: "3QcAYUrGoKv8tqg24VRB8TDUejmumLG8rL" },
   { network: "mainnet", address: "bc1q40aetdah8wcrsxvzl5qcft4svx696xsw0tnchd" },
   {
     network: "mainnet",
-    address: "bc1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcxswfvpysxf3qccfmv3",
+    address: "bc1p6pelce38k6dsjltcn53wqu0am5ucf3zrug8qx7x0gvdmay5s3ytsmnzj0u",
   },
-  { network: "testnet", address: "mipcBbFg9gLiKh81Kj8taadgoZiY1ZJRfn" },
-  { network: "testnet", address: "tb1qw508d6qejxtdg4y5r3aadgory0c5xw7kxpjzsx" },
+  { network: "testnet", address: "tb1qdhw6spmwavmu3fvajvsqx6gse9lvj4nhds606x" },
   {
     network: "testnet",
-    address: "tb1p84x2taadgovgnlpnxt9f39gm7r68gwtvllxqe5w2n5ru00s9aquslzggwq",
+    address: "tb1qklecm9q28lta8uevkeer729qu2te8jyzackw5c85dwd64t2630hsfv4u7v",
+  },
+  { network: "signet", address: "tb1qdhw6spmwavmu3fvajvsqx6gse9lvj4nhds606x" },
+  {
+    network: "signet",
+    address: "tb1qklecm9q28lta8uevkeer729qu2te8jyzackw5c85dwd64t2630hsfv4u7v",
+  },
+  { network: "regtest", address: "bcrt1qs758ursh4q9z627kt3pp5yysm78ddny6txaqgw" },
+  {
+    network: "regtest",
+    address: "bcrt1p0xlxvlhemja6c4dqv22uapctqupfhlxm9h8z3k2e72q4k9hcz7vqc8gma6",
+  },
+]
+
+const invalidAddresses = [
+  { network: "mainnet", address: "bc1q5a25l8l5r6kgestvadghsd88pjcj923mwed7ku4" },
+  {
+    network: "mainnet",
+    address: "bc1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcxswfvpysxf3qccfmv3",
```

Comment:
Why was this previously valid? (curiosity question)

Reply:
previously it was valid because we only had regex validation and not decode validation... you can check here https://cointools.org/valid-address-checker/?address=bc1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcxswfvpysxf3qccfmv3